pkg/aggregator/aggregator.go

```diff
@@ -346,7 +346,7 @@ func (agg *BufferedAggregator) registerSender(id check.ID) error {
 	if _, ok := agg.checkSamplers[id]; ok {
 		return fmt.Errorf("Sender with ID '%s' has already been registered, will use existing sampler", id)
 	}
-	agg.checkSamplers[id] = newCheckSampler()
+	agg.checkSamplers[id] = newCheckSampler(config.Datadog.GetDuration("check_sampler_bucket_expiry") * time.Second)
```

Comment:
Since you're pulling an integer from the config, use `GetInt` (`GetDuration` will read a duration-formatted string from the config):

```suggestion
	agg.checkSamplers[id] = newCheckSampler(time.Duration(config.Datadog.GetInt("check_sampler_bucket_expiry")) * time.Second)
```